Compilation failure on FreeBSD: no member named 'begin' in 'backward::StackTrace'

Originally reported by @yurivict in https://github.com/gap-system/gap/issues/6100

```
In file included from src/exception.cpp:33:
/usr/ports/math/gap/work/gap-4.15.0-beta1/pkg/semigroups/libsemigroups/extern/backward-cpp/backward.hpp:1337:25: error: no member named 'begin' in 'backward::StackTrace'
1337 |       load_addresses(st.begin(), static_cast<int>(st.size()));
|                      ~~ ^
src/exception.cpp:50:8: note: in instantiation of function template specialization 'backward::TraceResolverImplBase::load_stacktrace<backward::StackTrace>' requested here
50 |     tr.load_stacktrace(st);
|        ^
1 error generated.
```

Later I commented this line out and was able to build it, but the issue is still there.


Version: 4.15.0-beta1
clanng-19
FreeBSD 14.3

--- a/src/exception.cpp
+++ b/src/exception.cpp
@@ -59,22 +59,24 @@ namespace libsemigroups {
     st.load_here();
     backward::TraceResolver tr;
     tr.load_stacktrace(st);
-    for (size_t i = st.size() - 1; i-- > 0;) {
-      backward::ResolvedTrace trace = tr.resolve(st[i]);
-      size_t                  pos   = trace.object_function.find("(");
-      if (pos == std::string::npos) {
-        continue;
+    if (st.size() > 0) {
+      for (size_t i = st.size() - 1; i-- > 0;) {
+        backward::ResolvedTrace trace = tr.resolve(st[i]);
+        size_t                  pos   = trace.object_function.find("(");
+        if (pos == std::string::npos) {
+          continue;
+        }
+        std::string func(trace.object_function.begin(),
+                         trace.object_function.begin() + pos);
+        if (func.find("Catch") != std::string::npos
+            || func.find("C_A_T_C_H") != std::string::npos
+            || func.find("LibsemigroupsException") != std::string::npos
+            || func.find("pybind11") != std::string::npos
+            || func.find("libsemigroups") == std::string::npos) {
+          continue;
+        }
+        full_msg += "#" + std::to_string(i - 1) + " " + func + "\n";
       }
-      std::string func(trace.object_function.begin(),
-                       trace.object_function.begin() + pos);
-      if (func.find("Catch") != std::string::npos
-          || func.find("C_A_T_C_H") != std::string::npos
-          || func.find("LibsemigroupsException") != std::string::npos
-          || func.find("pybind11") != std::string::npos
-          || func.find("libsemigroups") == std::string::npos) {
-        continue;
-      }
-      full_msg += "#" + std::to_string(i - 1) + " " + func + "\n";
     }
     full_msg = "Simplified stack trace (most recent call last):\n" + full_msg;
     full_msg

